flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java

```diff
@@ -327,13 +376,14 @@ public T getClusterId() {
 	}
 
 	private <R, A extends AsynchronouslyCreatedResource<R>> R waitForResource(
-			final SupplierWithException<CompletableFuture<A>, IOException> resourceFutureSupplier)
+			final Supplier<CompletableFuture<A>> resourceFutureSupplier)
 				throws IOException, InterruptedException, ExecutionException, TimeoutException {
```

Comment:
`IOException` and `TimeoutException` not thrown.

Reply:
removed.. also made the method non-blocking